src/types.js

```diff
@@ -394,13 +549,18 @@ encode.MACSTRING = function(str, encoding) {
                 return undefined;
             }
         }
-
-        result.push(c);
+        result[i] = c;
```

Comment:
Here's a test that I did in Firefox and Chrome with results:
function timeJoin(){
	var start = new Date();
	var i=-1;
	var len  = 10000000;
	var arr = [];
	while(++i<len)
	{
		arr[i] = "a";
	}
	var str = arr.join("");
	var end = new Date();
	console.log(end.getTime()-start.getTime());
};
function timePush(){
	var start = new Date();
	var i=-1;
	var len  = 10000000;
	var arr = [];
	while(++i<len)
	{
		arr.push("a");
	}
	var str = arr.join("");
	var end = new Date();
	console.log(end.getTime()-start.getTime());
};

Chrome:
timeJoin()
822
timePush()
979

Firefox:
timeJoin()
184
timePush()
303

Unless there's something wrong with my test, push seems to be slower -- at least in Firefox. (Safari seems to be slightly slower with push as well.) Older browsers had a much bigger gap. I'm not sure about IE.